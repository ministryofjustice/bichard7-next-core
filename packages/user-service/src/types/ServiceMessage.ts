interface ServiceMessage {
  id: number
  message: string
  createdAt: Date
  incidentDate?: Date
}

export default ServiceMessage
