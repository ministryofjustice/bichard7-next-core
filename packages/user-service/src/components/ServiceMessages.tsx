import { format } from "date-fns"
import useCustomStyles from "styles/useCustomStyles"
import ServiceMessage from "types/ServiceMessage"
import GridColumn from "./GridColumn"
import GridRow from "./GridRow"
import Paragraph from "./Paragraph"

interface Props {
  messages: ServiceMessage[]
}

const ServiceMessages = ({ messages }: Props) => {
  const classes = useCustomStyles()

  return (
    <>
      {messages.length === 0 && <Paragraph>{"There are no service messages to display."}</Paragraph>}

      {messages.map((message, index) => (
        <GridRow key={String(index)}>
          <GridColumn width="full">
            <Paragraph className={`govuk-!-font-size-16 ${classes["service-message"]}`}>
              <time
                className="govuk-!-font-weight-bold govuk-!-font-size-14"
                aria-label="time"
                title={format(new Date(message.incidentDate || message.createdAt), "dd MMMM yyyy HH:mm")}
              >
                {format(new Date(message.incidentDate || message.createdAt), "dd MMMM yyyy")}
              </time>
              <br />
              {message.message}
            </Paragraph>
          </GridColumn>
        </GridRow>
      ))}
    </>
  )
}

export default ServiceMessages
