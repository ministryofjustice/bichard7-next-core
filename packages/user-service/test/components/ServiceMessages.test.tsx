import { render } from "@testing-library/react"
import ServiceMessages from "components/ServiceMessages"
import ServiceMessage from "types/ServiceMessage"

it("should render the component and match the snapshot when there are messages", () => {
  const messages: ServiceMessage[] = [
    { id: 1, message: "Message 1", createdAt: new Date("2021-09-28T10:12:13") },
    { id: 2, message: "Message 2", createdAt: new Date("2021-09-29T11:14:17") },
    {
      id: 3,
      message: "Message 3",
      createdAt: new Date("2021-09-30T17:45:23"),
      incidentDate: new Date("2021-10-10T16:30:00")
    }
  ]
  const { container } = render(<ServiceMessages messages={messages} />)

  expect(container).toMatchSnapshot()
})

it("should render the component and match the snapshot when there are no messages", () => {
  const { container } = render(<ServiceMessages messages={[]} />)

  expect(container).toMatchSnapshot()
})
