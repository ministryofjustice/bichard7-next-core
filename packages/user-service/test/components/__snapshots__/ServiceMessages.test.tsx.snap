// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`should render the component and match the snapshot when there are messages 1`] = `
<div>
  <div
    class="top-padding-0-2-5 govuk-grid-row"
  >
    <div
      class="govuk-grid-column-full"
    >
      <p
        class="govuk-body govuk-!-font-size-16 service-message-0-2-16"
      >
        <time
          aria-label="time"
          class="govuk-!-font-weight-bold govuk-!-font-size-14"
          title="28 September 2021 10:12"
        >
          28 September 2021
        </time>
        <br />
        Message 1
      </p>
    </div>
  </div>
  <div
    class="top-padding-0-2-5 govuk-grid-row"
  >
    <div
      class="govuk-grid-column-full"
    >
      <p
        class="govuk-body govuk-!-font-size-16 service-message-0-2-16"
      >
        <time
          aria-label="time"
          class="govuk-!-font-weight-bold govuk-!-font-size-14"
          title="29 September 2021 11:14"
        >
          29 September 2021
        </time>
        <br />
        Message 2
      </p>
    </div>
  </div>
  <div
    class="top-padding-0-2-5 govuk-grid-row"
  >
    <div
      class="govuk-grid-column-full"
    >
      <p
        class="govuk-body govuk-!-font-size-16 service-message-0-2-16"
      >
        <time
          aria-label="time"
          class="govuk-!-font-weight-bold govuk-!-font-size-14"
          title="10 October 2021 16:30"
        >
          10 October 2021
        </time>
        <br />
        Message 3
      </p>
    </div>
  </div>
</div>
`;

exports[`should render the component and match the snapshot when there are no messages 1`] = `
<div>
  <p
    class="govuk-body"
  >
    There are no service messages to display.
  </p>
</div>
`;
